### DISCUSSION TITLE
Errors about long cycle simulation

### DISCUSSION THREAD

**Discussion by YouHyin** [pybamm-team/PyBaMM#3085] [Q&A]
I created a function to compute the cycle life using the Prada2013 parameter sets. 
```python3
import pybamm
import numpy as np
from set_parameters import set_parameters

pybamm.set_logging_level("NOTICE")
parameter_values=set_parameters()
# Calculate stoichiometries at 100% SOC
parameter_values.set_initial_stoichiometries(1);
options = {
    "SEI": "ec reaction limited",
    "lithium plating": "irreversible",
    "thermal" : "lumped",
    }
spme = pybamm.lithium_ion.SPMe(options)
N_cycles=100

def compute_cycle_life(c1,c2,c3,soc1,soc2,soc3) :

    time_1=soc1*3600/c1
    time_2=soc2*3600/c2
    time_3=soc3*3600/c3

    experiment = pybamm.Experiment([
                                       (f"Discharge at 4C until 2.0V",
                                        f"Hold at 2.0V until C/50",
                                        "Rest for 5 minutes",
                                        f"Charge at {c1} C for {time_1} seconds",
                                        "Rest for 5 minutes",
                                        f"Charge at {c2} C for {time_2} seconds",
                                        "Rest for 5 minutes",
                                        f"Charge at {c2} C for {time_2} seconds",
                                        "Rest for 5 minutes",
                                        f"Hold at 3.6V until C/50",

                                        )
                                   ] * N_cycles,
                                   termination="95% capacity"

                                   )
    sim = pybamm.Simulation(spme, experiment=experiment, parameter_values=parameter_values)
    #solver = pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, mode="fast")
    sol=sim.solve()
    cycle_life=sol.summary_variables["Cycle number"]

    return cycle_life

print(compute_cycle_life(3.4,5,4.8,0.2,0.3,0.5))
```

However, when I ran this code, I got this error. 
`pybamm.expression_tree.exceptions.SolverError: Could not solve for summary variables, run `sim.solve(calc_esoh=False)` to skip this step

So I changed it to `sim.solve(calc_esoh=False)` and now I get this error and can't run it.
**raise KeyError(f"'{key}' not found. Best matches are {best_matches}")
KeyError: "'Capacity [A.h]' not found. Best matches are ['Measured capacity [A.h]', 'Throughput capacity [A.h]', 'Loss of capacity to SEI [A.h]']"**

What should I do to solve this situation??
I would be very grateful if you could let me know.

**Comment by brosaplanella** [comment]
The second error is because in order to access the `summary_variables` you need to have `calc_esoh=True`. For the first error it's hard to tell, could you share a minimum working example?

**Reply by YouHyin** [reply]
```

import pybamm
import numpy as np
from set_parameters import set_parameters


pybamm.set_logging_level("NOTICE")
parameter_values=set_parameters()

parameter_values.set_initial_stoichiometries(1);
options = {
    "SEI": "ec reaction limited",
    "lithium plating": "irreversible",
    "thermal" : "lumped",
    }
spme = pybamm.lithium_ion.SPMe(options)
N_cycles=10

def compute_cycle_life(c1,c2,c3,soc1,soc2,soc3) :
    time_1=soc1*3600/c1
    time_2=soc2*3600/c2
    time_3=soc3*3600/c3

    experiment = pybamm.Experiment([
                                       (f"Discharge at 4C until 2.0V",
                                        f"Hold at 2.0V until C/50",
                                        "Rest for 5 minutes",
                                        f"Charge at {c1} C for {time_1} seconds",
                                        "Rest for 5 minutes",
                                        f"Charge at {c2} C for {time_2} seconds",
                                        "Rest for 5 minutes",
                                        f"Charge at {c2} C for {time_2} seconds",
                                        "Rest for 5 minutes",
                                        f"Hold at 3.6V until C/50",

                                        )
                                   ] * N_cycles,

                                   )
    sim = pybamm.Simulation(spme, experiment=experiment, parameter_values=parameter_values)
    #solver = pybamm.CasadiSolver(atol=1e-6, rtol=1e-6, mode="fast")
    sol=sim.solve(calc_esoh=True)
    cyclelife=sol.summary_variables["Cycle number"]
    cycle_life=len(cyclelife)
    return cycle_life

print(compute_cycle_life(3.4,5,4.8,0.2,0.3,0.5))
```



First of all, thank you so much for your reply. I get this error when I run this code.How can I solve this problem?
![주석 2023-07-02 230208](https://github.com/pybamm-team/PyBaMM/assets/113442395/d6bcf206-8bac-42c3-9b23-8a7f982c0890)

**Reply by brosaplanella** [reply]
This example would not run as we do not have the `set_parameters` function. A minimum working example should be a single script with the minimum code to make the error appear (i.e. you should check which lines are not necessary to reproduce the error and remove it).

**Reply by Sparsh-Proj** [reply]
We are facing a similar issue. With "calc_esoh=False", we can not access the summary variables. But we'd like to get summary variables for plotting purposes. 
With "calc_esoh=True", (Which is the default value, nonetheless), the code fails and throws a SolverError as follows:
```SolverError: Could not solve for summary variables, run `sim.solve(calc_esoh=False)` to skip this step```

Link to minimum working code examples' google colab: https://colab.research.google.com/drive/11xUA7_o9avRH6ka3RgN1qEQAnAuqPtUj?usp=sharing

I've attached the same code below as well:

```
%pip install pybamm -q

import pybamm

pybamm.set_logging_level("NOTICE")

model = pybamm.lithium_ion.DFN()

params = pybamm.ParameterValues("Prada2013")

exp = pybamm.Experiment(
    [
        ("Discharge at C/10 for 10 hours or until 2.8 V",
        "Rest for 1 hour",
        "Charge at 5 A until 3.5 V",
        "Hold at 3.5 V until 50 mA",
        "Rest for 1 hour"),
    ] * 10
)

sim = pybamm.Simulation(
                          model, 
                          experiment = exp, 
                          parameter_values = params
                       )

# sol = sim.solve()

# Throws following error: SolverError: Could not solve for summary variables, run `sim.solve(calc_esoh=False)` to skip this step

# Bypassing the error:

sol = sim.solve(calc_esoh = False)

pybamm.plot_summary_variables(sol)

# Throws following error: KeyError: "'Capacity [A.h]' not found. Best matches are ['Measured capacity [A.h]', 'Throughput capacity [A.h]', 'Loss of capacity to SEI [A.h]']"
```

**Reply by brosaplanella** [reply]
I think the issues are closely related to the Prada parameter set. Some of the values did not match the paper, and this has now been fixed in the develop branch (https://github.com/pybamm-team/PyBaMM/pull/3096). You can try to install from source and see if it works.

**Reply by YouHyin** [reply]
Hi, I saw (https://github.com/pybamm-team/PyBaMM/pull/3096) and I don't quite understand: are you asking me to install pybamm version 23.5?? If so, can I install it via this command? (conda install -c conda-forge pybamm=23.5) I'm having trouble installing it, how can I fix it? Thank you very much.

**Reply by brosaplanella** [reply]
I am not familiar with conda (I use pip), but I guess the instructions here should work: https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-pkgs.html#updating-packages
